jdk/make/CompileLaunchers.gmk

```diff
@@ -256,14 +256,36 @@ $(JDK_OUTPUTDIR)/bin$(OUTPUT_SUBDIR)/java$(EXE_SUFFIX): $(BUILD_LAUNCHER_java)
 	$(RM) $@
 	$(CP) $(JDK_OUTPUTDIR)/objs/java_objs$(OUTPUT_SUBDIR)/java$(EXE_SUFFIX) $@
 
+ifeq ($(ZIP_DEBUGINFO_FILES), true)
+  DEBUGINFO_EXT := .diz
+else ifeq ($(OPENJDK_TARGET_OS), macosx)
+  DEBUGINFO_EXT := .dSYM
+else ifeq ($(OPENJDK_TARGET_OS), windows)
+  DEBUGINFO_EXT := .pdb
+else
+  DEBUGINFO_EXT := .debuginfo
+endif
+
+$(JDK_OUTPUTDIR)/bin$(OUTPUT_SUBDIR)/java$(DEBUGINFO_EXT): $(BUILD_LAUNCHER_java)
+	$(MKDIR) -p $(@D)
+	$(RM) $@
+	$(CP) -R $(JDK_OUTPUTDIR)/objs/java_objs$(OUTPUT_SUBDIR)/java$(DEBUGINFO_EXT) $@
+
 BUILD_LAUNCHERS += $(JDK_OUTPUTDIR)/bin$(OUTPUT_SUBDIR)/java$(EXE_SUFFIX)
+ifeq ($(ENABLE_DEBUG_SYMBOLS), true)
+  ifneq ($(POST_STRIP_CMD), )
+    ifneq ($(STRIP_POLICY), no_strip)
+      BUILD_LAUNCHERS += $(JDK_OUTPUTDIR)/bin$(OUTPUT_SUBDIR)/java$(DEBUGINFO_EXT)
+    endif
+  endif
+endif
 
 ifeq ($(OPENJDK_TARGET_OS):$(ENABLE_DEBUG_SYMBOLS), macosx:true)
   # Because the java launcher is built in a non-standard directory,
   # we need to copy debug information so it can be included in a debug image.
   $(JDK_OUTPUTDIR)/bin$(OUTPUT_SUBDIR)/java.dSYM/Contents/Info.plist : $(BUILD_LAUNCHER_java)
 	$(MKDIR) -p $(@D)
-	$(RM) $@
+	$(RM) -R $@
```

Comment:
This use of `-R` is inappropriate (the target is a file).

Reply:
It is to fix
```
20:01:43  rm: /Users/jenkins/workspace/Build_JDK8_x86-64_mac_Personal/build/macosx-x86_64-normal-server-release/jdk/bin/java.dSYM: is a directory
20:01:43  make[2]: *** [/Users/jenkins/workspace/Build_JDK8_x86-64_mac_Personal/build/macosx-x86_64-normal-server-release/jdk/bin/java.dSYM] Error 1
```
`java.dSYM` is a directory.

Reply:
This rule describes how to make .../java.dSYM/Contents/Info.plist, which is a file.
We want to minimize the changes to openjdk code (the problem is not here anyway).

Reply:
Removed this `-R`, and verified in a personal build. (local build works as before).

Also merged with latest `openj9` branch.

Please have another look.